config/convert.go

```diff
@@ -0,0 +1,267 @@
+
+	"github.com/alecthomas/units"
+	"github.com/coreos/ignition/config/types"
+	"github.com/vincent-petithory/dataurl"
+)
+
+const (
+	BYTES_PER_SECTOR = 512
+)
+
+func ConvertAs2_0_0(in Config) (types.Config, error) {
+	out := types.Config{
+		Ignition: types.Ignition{
+			Version: types.IgnitionVersion{Major: 2, Minor: 0},
+		},
+	}
+
+	for _, ref := range in.Ignition.Config.Append {
+		newRef, err := convertConfigReference(ref)
+		if err != nil {
+			return types.Config{}, err
+		}
+		out.Ignition.Config.Append = append(out.Ignition.Config.Append, newRef)
+	}
+
+	if in.Ignition.Config.Replace != nil {
+		newRef, err := convertConfigReference(*in.Ignition.Config.Replace)
+		if err != nil {
+			return types.Config{}, err
+		}
+		out.Ignition.Config.Replace = &newRef
+	}
+
+	for _, disk := range in.Storage.Disks {
+		newDisk := types.Disk{
+			Device:    types.Path(disk.Device),
+			WipeTable: disk.WipeTable,
+		}
+
+		for _, partition := range disk.Partitions {
+			size, err := convertPartitionDimension(partition.Size)
+			if err != nil {
+				return types.Config{}, err
+			}
+			start, err := convertPartitionDimension(partition.Start)
+			if err != nil {
+				return types.Config{}, err
+			}
+
+			newDisk.Partitions = append(newDisk.Partitions, types.Partition{
+				Label:    types.PartitionLabel(partition.Label),
+				Number:   partition.Number,
+				Size:     size,
+				Start:    start,
+				TypeGUID: types.PartitionTypeGUID(partition.TypeGUID),
+			})
+		}
+
+		out.Storage.Disks = append(out.Storage.Disks, newDisk)
+	}
+
+	for _, array := range in.Storage.Arrays {
+		newArray := types.Raid{
+			Name:   array.Name,
+			Level:  array.Level,
+			Spares: array.Spares,
+		}
+
+		for _, device := range array.Devices {
+			newArray.Devices = append(newArray.Devices, types.Path(device))
+		}
+
+		out.Storage.Arrays = append(out.Storage.Arrays, newArray)
+	}
+
+	for _, filesystem := range in.Storage.Filesystems {
+		newFilesystem := types.Filesystem{
+			Name: filesystem.Name,
+			Path: func(p types.Path) *types.Path {
+				if p == "" {
+					return nil
+				}
+
+				return &p
+			}(types.Path(filesystem.Path)),
+		}
+
+		if filesystem.Mount != nil {
+			newFilesystem.Mount = &types.FilesystemMount{
+				Device: types.Path(filesystem.Mount.Device),
+				Format: types.FilesystemFormat(filesystem.Mount.Format),
+			}
+
+			if filesystem.Mount.Create != nil {
+				newFilesystem.Mount.Create = &types.FilesystemCreate{
+					Force:   filesystem.Mount.Create.Force,
+					Options: types.MkfsOptions(filesystem.Mount.Create.Options),
+				}
+			}
+		}
+
+		out.Storage.Filesystems = append(out.Storage.Filesystems, newFilesystem)
+	}
+
+	for _, file := range in.Storage.Files {
+		newFile := types.File{
+			Filesystem: file.Filesystem,
+			Path:       types.Path(file.Path),
+			Mode:       types.FileMode(file.Mode),
+			User:       types.FileUser{Id: file.User.Id},
+			Group:      types.FileGroup{Id: file.Group.Id},
+		}
+
+		if file.Contents.Inline != "" {
```

Comment:
```
files:
    - path: /etc/kubernetes/empty
      filesystem: somefs
      mode: 0644
      contents:
        inline: |
```

Adding that to a fuze config winds up dropping to an emergency shell on boot because no Inline contents falls through to the URL scheme below.

createFiles: fetching file /etc/kubernetes/empty: unsupported source scheme
failed to create files: failed to resolve file /etc/kubernetes/empty

In one case, an empty file like this was used for the side-effect of ensuring the `/etc/kubernetes` directory existed. Its easily worked around by just adding some dummy file contents for now.